extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/pipeline/HttpRequestParams.java

```diff
@@ -81,17 +81,18 @@ private RequestBody createRequestBody(@Nullable Supplier<InputStream> bodySuppli
      *
      * @return The URL.
      */
-    private String toUrl() {
-        var url = baseUrl;
+    private HttpUrl toUrl() {
+        var parsed = HttpUrl.parse(baseUrl);
+        Objects.requireNonNull(parsed, "Failed to parse baseUrl: " + baseUrl);
+        var builder = parsed.newBuilder();
         if (path != null && !path.isBlank()) {
-            url += "/" + path;
+            builder.addPathSegments(path);
         }
-        if (queryParams != null && !queryParams.isBlank()) {
-            url += "?" + queryParams;
-        }
-        return url;
+        queryParams.forEach(builder::addQueryParameter);
```

Comment:
withouth parsing the query params and then passing them to the builder why don't use the `HttpUrl.Builder.encodedUrl`, that would avoid the parse work on our side.

Reply:
Using `encodedQuery` would override the potential query parameters already provided in the baseUrl.

Reply:
I think the query params should be either on the `baseUrl` or in the `queryParams` , not in both. At least that's not currently supported.

Reply:
We currently have a use-case in Eona-X wherein some query parameters are internal to the API (for example some API filters with some internal codes that should not be exposed to the consumer). 
On the other hand, the `queryParams` field convey the query parameters provided by the consumer through the proxy. 

So for me we should support both.

Reply:
Could you please check if latest commit fits better? https://github.com/eclipse-edc/Connector/pull/2381/commits/977e06943da7b3cf975f7b6189f4311ba596ec2d